packages/nuxt-module/plugins/i18n-cookies.js

```diff
@@ -24,13 +24,14 @@ const i18nCookiesPlugin = ({ $cookies, i18n, app, redirect }) => {
   };
 
   const getCurrencyByLocale = (locale) =>
-    i18n.numberFormats?.[locale]?.currency?.currency
+    app.$config.defaultCurrency
+    || i18n.numberFormats?.[locale]?.currency?.currency
     || i18nOptions.currency
     || (i18nOptions.currencies.length && i18nOptions.currencies[0].name);
```

Comment:
I am not sure if I understand, but this function is supposed to get currency by locale and you added the logic to return default currency regardless of the locale parameter value. This is a surprising behavior. We should think if we want to set default value or locale-correlated value, or both. I mean, even if we want to keep the logic you introduced, the naming convention here is very misleading when it comes to understanding the code. Also I understand that this is a no bc change.

Reply:
I think the easiest solution would be to simply remove `byLocale` from the function name.

Reply:
WDYT @bartoszherba ?

Reply:
Then the parameter for the function is a big unknown. Everyone will expect to get the currency based on that parameter. Maybe it should be resolved to null by default and only in this case, we should pull the default currency.

Reply:
In that case, what should make of the previous implementation? Here we also have only one of the three cases leveraging the `locale` param and the remaining two just pick the currency from `i18nOptions` without using the `locale` param at all.

```
  const getCurrencyByLocale = (locale) =>
    i18n.numberFormats?.[locale]?.currency?.currency
    || i18nOptions.currency
    || (i18nOptions.currencies.length && i18nOptions.currencies[0].name);
```

Reply:
@bartoszherba I've reverted `getCurrencyByLocale` to its initial state, could you please have a look once more?